src/index.php

```diff
@@ -1,133 +1,44 @@
-
-$islandoraTransactionService = new TransactionServiceProvider;
-
-$app->register($islandoraTransactionService);
-$app->mount("/islandora", $islandoraTransactionService);
+// TODO: Not register all template directories right now.
+$app->register(new \Silex\Provider\TwigServiceProvider(), array(
+  'twig.path' => array(
+    __DIR__ . 'CollectionService/templates',
+  ),
+));
 
-$islandoraCollectionService = new CollectionServiceProvider;
+$islandoraCollectionServiceProvider = new CollectionServiceProvider;
 
-$app->register(
-    $islandoraCollectionService,
-    array(
-      'UuidGenerator' => new UuidGenerator(),
-    )
+$basepath = array(
+  'islandora.BasePath' => __DIR__,
 );
-$app->mount("/islandora", $islandoraCollectionService);
+
+$app->register($islandoraCollectionServiceProvider, $basepath);
+$app->mount("/islandora", $islandoraCollectionServiceProvider);
 
 /**
- * Convert returned Guzzle responses to Symfony responses.
+ * Convert returned Guzzle responses to Symfony responses, type hinted.
  */
-
-$app->view(
-    function (ResponseInterface $psr7) {
-        return new Response($psr7->getBody(), $psr7->getStatusCode(), $psr7->getHeaders());
-    }
-);
-
-
-$app->after(
-    function (Request $request, Response $response) use ($app) {
-        $response->headers->set('X-Powered-By', 'Islandora Collection REST API v'
-        .$app['config']['islandora']['apiVersion'], true); //Nice
-    }
-);
-
-//Common error Handling
-$app->error(
-    function (\EasyRdf_Exception $e, $code) use ($app) {
-        if ($app['debug']) {
-            return;
-        }
-
-        return new response(sprintf('RDF Library exception', $e->getMessage(), $code), $code);
-    }
-);
-$app->error(
-    function (\Symfony\Component\HttpKernel\Exception\HttpException $e, $code) use ($app) {
-        if ($app['debug']) {
-            return;
-        }
-
-        return new response(
-            sprintf(
-                'Islandora Collection Service exception: %s / HTTP %d response',
-                $e->getMessage(),
-                $code
-            ),
-            $code
-        );
-    }
-);
-$app->error(
-    function (\Symfony\Component\HttpKernel\Exception\NotFoundHttpException $e, $code) use ($app) {
-        if ($app['debug']) {
-            return;
-        }
-        //Not sure what the best "verbose" message is
-        return new response(
-            sprintf(
-                'Islandora Collection Service exception: %s / HTTP %d response',
-                $e->getMessage(),
-                $code
-            ),
-            $code
-        );
-    }
-);
-$app->error(
-    function (\Exception $e, $code) use ($app) {
-        if ($app['debug']) {
-            return;
-        }
-
-        return new response(
-            sprintf(
-                'Islandora Collection Service uncatched exception: %s %d response',
-                $e->getMessage(),
-                $code
-            ),
-            $code
-        );
-    }
-);
```

Comment:
Leave in all the above `$app->error` functions. They make it so we can pass a pretty error if you are not using `$app[debug] = true;`